report/4.Design/RTS.tex

```diff
@@ -6,7 +6,74 @@ \subsection{Scheduling}\label{Design:Scheduling}
 When choosing a scheduling method for the system, multiple methods were considered. 
 Both fixed priority scheduling and cyclic executive were considered for the system as both of them would be good at handling the tasks of the system with their deadlines.
 
-Cyclic executive was chosen due the the fast execution time of the tasks, and that preemption and context switching would exceed the time for simply completing the previous task. 
-It was reasoned that since all tasks should run regularly, and execute fast enough to run within the major cycle the cyclic execution method should be perfect. 
+Due to the fast execution time of the tasks, and that preemption and context switching would exceed the time for simply completing the previous task.
+At first it was at deemed obvious to use the cyclic executive model as all tasks should run regularly, and that they execute fast enough to run within the major cycle the cyclic execution method should be perfect. 
+However, should the execution time of a single task exceed one millisecond, the usage of cyclic executive is no longer viable, as the task to keep the USB connection alive is required to be executed once every millisecond.
+In this case, fixed-priority scheduling is a better alternative, as it allows the scheduler to pre-empt the task in order to run the USB related task.
 
-The cyclic executive method is a common way to implement hard real-time systems due to its simplicity, deterministic behavior and relative ease of implementation\cite{CyclicExecutionKimLarsen}.
\ No newline at end of file
+Both the fixed priority and cyclic executive methods are a common ways to implement hard real-time systems due to their simplicity, deterministic behavior and relative ease of implementation\cite{CyclicExecutionKimLarsen}.
+
+In order to ensure that the system is schedulable, an UPPAAL model of the scheduler and the tasks was created. 
+
+\subsubsection{UPPAAL}
+UPPAAL is an integrated tool environment for modeling, validation and verification of real-time systems modeled as networks of timed automata\cite{UPPAALWebsite}. 
+Using the UPPAAL tool, a model of the tasks of the system was constructed. 
+
+\paragraph{Templates}
+When creating an UPPAAL project, a templates must be made, which is a model that describes a system or a part of a system.  
```

Comment:
```suggestion
When creating an UPPAAL project, a templates must be made.
Templates are a model that describes a system or a part of a system.  
```

Reply:
Burde det ikke være part of a system, siden der nedenunder skrives at de er brugt til at lave et helt system.